eXo/Sources/Controllers/HomePage/HomePageViewController.swift

```diff
@@ -603,3 +622,134 @@ extension HomePageViewController:WKDownloadDelegate {
     }
     
 }
+
+extension HomePageViewController:JitsiMeetViewDelegate{
+    
+    // Open Jitis Meet.
+
+    func openJitsiMeet(serverUrl:String,room:String) {
+        let jitsiServerURL = serverUrl.serverDomainWithProtocolAndPort! + Config.eXoJitsiWebServer
+        cleanUp()
+        self.getJWT_token(serverUrl: serverUrl) { jwt in
+            // create and configure jitsimeet view
+            DispatchQueue.main.async {
+                let jitsiMeetView = JitsiMeetView()
+                jitsiMeetView.delegate = self
+                self.jitsiMeetView = jitsiMeetView
+                let options = JitsiMeetConferenceOptions.fromBuilder { (builder) in
+                    builder.serverURL = URL(string:jitsiServerURL)
+                    builder.token = jwt
+                    builder.room = room
+                    builder.setFeatureFlag("welcomepage.enabled", withValue: false)
+                }
+                jitsiMeetView.join(options)
+                // Enable jitsimeet view to be a view that can be displayed
+                // on top of all the things, and let the coordinator to manage
+                // the view state and interactions
+                self.pipViewCoordinator = PiPViewCoordinator(withView: jitsiMeetView)
+                self.pipViewCoordinator?.configureAsStickyView(withParentView: self.view)
+                // animate in
+                jitsiMeetView.alpha = 0
+                self.pipViewCoordinator?.show()
+            }
+        }
+    }
+    
+    func openJitsiMeet(webview:WKWebView,url:URL){
+            cleanUp()
+            webViewDidClose(webview)
+            let username = PushTokenSynchronizer.shared.username!
```

Comment:
Couldn't we get the username from the main app instead of the PushTokenSynchronier ?